Library/Homebrew/cmd/update.sh

```diff
@@ -390,6 +390,22 @@ EOS
     fi
   fi
 
+  if [[ -f "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-core/.git/shallow" &&
+      -f "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-cask/.git/shallow" ]]
+  then
+    odie <<EOS
+homebrew-core and homebrew-cask are shallow clones. To \`brew update\` first run:
+  git -C "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-core" fetch --unshallow
+  git -C "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-cask" fetch --unshallow
+This restriction has been made on GitHub's request because updating shallow
+clones is an extremely expensive operation due to the tree layout and traffic of
+Homebrew/homebrew-core and Homebrew/homebrew-cask. We don't do this for you
+automatically to avoid repeatedly performing an expensive unshallow operation in
+CI systems (which should instead be fixed to not use shallow clones). Sorry for
+the inconvenience!
+EOS
+  fi
+
```

Comment:
This code doesn't look kosher. If we merge this "as is", we'd have the same message repeated three times.
I think we should do something like this:
```sh
[[ -f "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-core/.git/shallow" ]] && SHALLOW_CORE=1
[[ -f "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-cask/.git/shallow" ]] && SHALLOW_CASK=1

if [[ -n $SHALLOW_CORE || -n $SHALLOW_CASK ]]
then
  odie <<EOS
${SHALLOW_CORE+homebrew-core is a shallow clone.}
${SHALLOW_CASK+homebrew-cask is a shallow clone.}
To \`brew update\`, first run:
  ${SHALLOW_CORE+git -C "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-core" fetch --unshallow}
  ${SHALLOW_CASK+git -C "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-cask" fetch --unshallow}
This restriction has been made on GitHub's request because updating shallow
clones is an extremely expensive operation due to the tree layout and traffic of
Homebrew/homebrew-core and Homebrew/homebrew-cask. We don't do this for you
automatically to avoid repeatedly performing an expensive unshallow operation in
CI systems (which should instead be fixed to not use shallow clones). Sorry for
the inconvenience!
EOS
fi
```

Reply:
As far as I can tell in my testing "If we merge this "as is", we'd have the same message repeated three times." Something (odie? `set -e`?) causes this to not display multiple messages. 

That said I would like your idea better so cool to go with that.

What I would really like to understand is why only one message gets displayed at a time. searching for odie found me a great lasagna recipe but not what is causing the first message to end the output.

Reply:
> As far as I can tell in my testing "If we merge this "as is", we'd have the same message repeated three times." is not true.

I meant in code, not "displayed to the user" 👍🏻